launch_ros/launch_ros/actions/node.py

```diff
@@ -461,6 +471,11 @@ def execute(self, context: LaunchContext) -> Optional[List[Action]]:
                     'launch context'.format(node_name_count, self.node_name)
                 )
 
+        # Delete temporary parameter yaml file
+        if param_file_cleanup and param_file_path is not None:
```

Comment:
The file cannot be removed here, as the executable may not have parsed it yet.
We could delete the files in an [OnExit](https://github.com/ros2/launch/blob/4370b2b829946cd8c5ec5a76cbb3bb6a70c95e97/launch/launch/actions/execute_process.py#L106) event (that is executed when the process is finished).

Reply:
a30d634

Reply:
You need to register an event handler, e.g.: [here](https://github.com/ros2/launch/blob/4370b2b829946cd8c5ec5a76cbb3bb6a70c95e97/launch/launch/actions/execute_process.py#L831).

The launch event loop will trigger the registered `OnExit` event handler when the process finishes.